File: sdk/scala-sdk-protobuf/src/main/scala/kalix/scalasdk/workflow/WorkflowOptions.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.scalasdk.workflow

import kalix.scalasdk.impl.ComponentOptions

/** Workflow options. */
trait WorkflowOptions extends ComponentOptions {

  def withForwardHeaders(headers: Set[String]): WorkflowOptions
}
object WorkflowOptions {
  val defaults: WorkflowOptions =
    WorkflowOptionsImpl(Set.empty)

  private[kalix] final case class WorkflowOptionsImpl(forwardHeaders: Set[String]) extends WorkflowOptions {

    override def withForwardHeaders(headers: Set[String]): WorkflowOptionsImpl =
      copy(forwardHeaders = headers)
  }
}

File: sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/workflow/WorkflowOptions.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.javasdk.workflow;

import kalix.javasdk.impl.workflow.WorkflowOptionsImpl;

public interface WorkflowOptions extends kalix.javasdk.impl.ComponentOptions {

  static WorkflowOptions defaults() {
    return WorkflowOptionsImpl.defaults();
  }
}
